packages/pros/Cargo.toml

```diff
@@ -18,17 +18,21 @@ rust-version = "1.75.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-lazy_static = { version = "1.4.0", features = ["spin_no_std"] }
-spin = "0.9.8"
-pros-sys = { version = "0.6", path = "../pros-sys", features = ["xapi"] }
-snafu = { version = "0.8.0", default-features = false, features = [
-    "rust_1_61",
-    "unstable-core-error",
-] }
-no_std_io = { version = "0.6.0", features = ["alloc"] }
-futures = { version = "0.3.28", default-features = false, features = ["alloc"] }
-async-task = { version = "4.5.0", default-features = false }
-waker-fn = "1.1.1"
+pros-sync = { version = "0.1.0", path = "../pros-sync", optional = true }
+pros-async = { version = "0.1.0", path = "../pros-async", optional = true }
+pros-devices = { version = "0.1.0", path = "../pros-devices", optional = true }
+pros-panic = { version = "0.1.0", path = "../pros-panic", optional = true }
+pros-core = { version = "0.1.0", path = "../pros-core", optional = true }
 
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-dlmalloc = { version = "0.2.4", features = ["global"] }
+[features]
+default = ["async", "sync", "devices", "panic", "display_panics", "core"]
+
+core = ["dep:pros-core"]
```

Comment:
I feel like core shouldnt be an optional feature since everything depends on it anyways. I can't imagine any point in using `pros` if you don't also want `pros_core`.

Reply:
I think it should still be a future. If you want to make a supporting library for pros-rs and import pros crates using the metapackage, it's perfectly reasonable to not need `pros-core`.

Reply:
All other crates in the meta package depend on core, so the only use case I can imagine would be the rather useless scenario of using the meta package with zero features enabled (in which case it would export nothing).

Reply:
Just because all pros crates depend on pros-core doesn't mean you can't have pros-devices enabled but not pros core for example. Even though pros-devices depends on pros-core, you might not want the pros-core feature. If you want a custom allocator for example you wont want to have the pros-core dependency, but you might want to still use the pros metapackage.